framework-support/jobrunr-spring-boot-2-starter/src/main/java/org/jobrunr/scheduling/AsyncJobPostProcessor.java

```diff
@@ -0,0 +1,49 @@
+package org.jobrunr.scheduling;
+
+import org.jobrunr.jobs.annotations.Job;
+import org.jobrunr.jobs.annotations.AsyncJob;
+import org.springframework.aop.framework.ProxyFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.BeanFactoryAware;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.stereotype.Component;
+
+import java.lang.reflect.InvocationTargetException;
+
+@Component
+public class AsyncJobPostProcessor implements BeanPostProcessor, BeanFactoryAware {
+    BeanFactory beanFactory;
+
+    @Override
+    public Object postProcessBeforeInitialization(Object bean, String beanName) {
+        if (bean.getClass().isAnnotationPresent(AsyncJob.class)) {
+            try {
+                return applyJobEnhancement(bean);
+            } catch (InstantiationException | IllegalAccessException | NoSuchMethodException | InvocationTargetException e) {
+                throw new RuntimeException(e);
+            }
+        }
+        return bean;
+    }
+
+    public Object applyJobEnhancement(Object bean) throws InstantiationException, IllegalAccessException, NoSuchMethodException, InvocationTargetException {
```

Comment:
Are these exceptions thrown from the code inside this method? As I don't see it. So they could be removed as well as the catch from the `postProcessBeforeInitialization`.

Reply:
Fixed